pkg/secrets-store/utils.go

```diff
@@ -0,0 +1,727 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secretsstore
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var (
+	secretProviderClassGvk = schema.GroupVersionKind{
+		Group:   "secrets-store.csi.x-k8s.io",
+		Version: "v1alpha1",
+		Kind:    "SecretProviderClassList",
+	}
+)
+
+// getProviderPath returns the absolute path to the provider binary
+func (ns *nodeServer) getProviderPath(goos string, providerName string) string {
+	if goos == "windows" {
+		return normalizeWindowsPath(fmt.Sprintf(`%s\%s\provider-%s.exe`, ns.providerVolumePath, providerName, providerName))
+	}
+	return fmt.Sprintf("%s/%s/provider-%s", ns.providerVolumePath, providerName, providerName)
+}
+
+func normalizeWindowsPath(path string) string {
+	normalizedPath := strings.Replace(path, "/", "\\", -1)
+	if strings.HasPrefix(normalizedPath, "\\") {
+		normalizedPath = "c:" + normalizedPath
+	}
+	return normalizedPath
+}
+
+// getMountedFiles returns all the mounted files
+func getMountedFiles(targetPath string) ([]string, error) {
+	// loop thru all the mounted files
+	files, err := filepath.Glob(filepath.Join(targetPath, "*"))
+	if err != nil {
+		log.Errorf("failed to list all files in target path %s, err: %v", targetPath, err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	return files, nil
+}
+
+// getPodUIDFromTargetPath returns podUID from targetPath
+func getPodUIDFromTargetPath(goos string, targetPath string) string {
+
+	var parts []string
+	if goos == "windows" {
+		if strings.Contains(targetPath, `\var\lib\kubelet\pods`) {
```

Comment:
I think we need to also handle the case for `\\` in windows target path. I think we can check the prefix in the target path - `\` or `\\` and use the appropriate one as the separator?